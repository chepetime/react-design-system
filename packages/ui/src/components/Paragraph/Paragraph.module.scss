@use "./../../styles" as styles;

:root {
  --paragraph-background-color: #{styles.$pink-500};
}

:global .dark {
  --paragraph-background-color: #{styles.$indigo-500};
}

.Paragraph {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--paragraph-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
