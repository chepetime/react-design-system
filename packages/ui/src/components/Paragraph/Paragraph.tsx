import React from "react";
import classnames from "classnames";
import css from "./Paragraph.module.scss";

export interface ParagraphProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Paragraph = React.forwardRef<HTMLDivElement, ParagraphProps>(
  (props: ParagraphProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Paragraph)} ref={ref}>
        <p>Paragraph component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Paragraph.displayName = "Paragraph";
