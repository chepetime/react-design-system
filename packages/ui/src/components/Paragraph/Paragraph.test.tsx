import React from "react";import { render } from "@testing-library/react";
import { Paragraph } from "./Paragraph";

describe("Paragraph", () => {
  test("renders the Paragraph component", () => {
    render(<Paragraph />);
  });
});
