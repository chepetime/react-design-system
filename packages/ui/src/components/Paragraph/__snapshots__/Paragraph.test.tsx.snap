// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph matches snapshot 1`] = `
<div>
  <div
    class="Paragraph"
  >
    <p>
      Paragraph component working!
    </p>
    <div />
  </div>
</div>
`;
